src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs

```diff
@@ -218,24 +218,127 @@ public async Task When_StartingFrom_AnimatedValue()
 				To = -50,
 				Duration = new Duration(TimeSpan.FromSeconds(5)),
 			}.BindTo(translate, nameof(translate.Y));
-			animation1.Begin();
+			animation1.Wrap().Begin();
 			await Task.Delay(125);
 
 			// ~125ms into a 5s animation where the value is animating from 50 to -50,
 			// the value should be still positive.
-			// note: On android, the value will be scaled by ViewHelper.Scale, but the assertion will still hold true.
 			var y = GetTranslateY(translate, isStillAnimating: true);
 			Assert.IsTrue(y > 0, $"Expecting Translate.Y to be still positive: {y}");
 		}
 
-		private double GetTranslateY(TranslateTransform translate, bool isStillAnimating = false) =>
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_StartingFrom_AnimatingValue() // value from mid animation
+		{
+			var translate = new TranslateTransform();
+			var border = new Border()
+			{
+				Background = new SolidColorBrush(Colors.Pink),
+				Margin = new Thickness(0, 50, 0, 0),
+				Width = 50,
+				Height = 50,
+				RenderTransform = translate,
+			};
+			WindowHelper.WindowContent = border;
+			await WindowHelper.WaitForLoaded(border);
+			await WindowHelper.WaitForIdle();
+
+			translate.Y = 50;
+			await WindowHelper.WaitForIdle();
+			await Task.Delay(1000);
+			var threshold = GetTranslateY(translate); // snapshot the value
+
+			// Start an animation. Its animating value will serve as
+			// the inferred starting value for the next animation.
+			var animation0 = new DoubleAnimation
+			{
+				From = 100,
+				To = 105,
+				Duration = new Duration(TimeSpan.FromSeconds(5)),
+			}.BindTo(translate, nameof(translate.Y));
+			animation0.Wrap().Begin();
+			await Task.Delay(125);
```

Comment:
Even if other tests are using the same pattern, are you able to test it without delays? Time based tests have about 90% of chance to fail randomly on the CI :/

Reply:
the goal is to capture the value on the flight, hence a super large duration (5s) and only waiting briefly (125ms)
so long as we are in that 5s window, it is good
we are re-animating this value again in the next part, wherever it snapshots in the 100~105 doesnt matter
as the failing condition was checking for a starting value of 0